web/src/components/pages/contribution/speak/speak.tsx

```diff
@@ -521,6 +517,7 @@ class SpeakPage extends React.Component<Props, State> {
                     className={getTrackClass('fs', 'exit-submit-clips')}
                     onClick={() => {
                       if (this.upload()) onConfirm();
+                      this.resetAndGoHome();
```

Comment:
Thanks for finding a pre-existing method to solve the problem! Unfortunately, this will causes a few additional bugs. Your code sends users back to the homepage no matter what. If someone wanted to navigate elsewhere (eg. if they clicked the "Listen" button), or if the upload failed, they would still be directed to the homepage.

The actual bug that we want to solve here, I think, is that `onConfirm()` isn't being called when it should be. It could have something to do with the Terms & Conditions modal popping up?

Reply:
I encountered this  too 
>  Your code sends users back to the homepage no matter what. If someone wanted to navigate elsewhere 

Since the `onConfirm` is working can be cross-checked from the `Exit & Delete clips` button.
So one thing is clear that the function is working. 
> The actual bug that we want to solve here, I think, is that `onConfirm()` isn't being called when it should be. It could have something to do with the Terms & Conditions modal popping up?

Maybe this is happening here, the if condition is checking the value of `this.upload()` on the click of `submit Clips` button which is declared initially false. So that may be the reason for the `onConfirm()` not be called.
Your views?

Reply:
Could be! It's likely some sort of race condition like that. Let me know if you figure it out :)

Reply:
Calling the `this.upload()` on click and waiting for it to get completed.
Then checking that the clip is uploaded or not will make sense?
This solution would be acceptable?

Reply:
Yeah – `this.upload` should probably be asynchronous, honestly. This is the issue in the `upload` method:

```
    if (!hasAgreed && !(user.privacyAgreed || user.account)) {
      this.setState({ showPrivacyModal: true });
      return false;
    }
```
^ meaning `onConfirm` will never run if the user hasn't accepted the privacy agreement yet.

We should wait for the privacy modal to be accepted, then trigger the redirect. Ideally this would all be asynchronous (using callbacks, promises, or async/await). Practically, that might be a huge code change. So I'd also consider a fix that saves `onConfirm` to state until the privacy modal is accepted.

Reply:
>  So I'd also consider a fix that saves `onConfirm` to state until the privacy modal is accepted.

That means, make another function(which will do the same work as of onConfirm), save into the state and call it after the privacy modal is accepted, that will close the navigation prompt. 
Correct me if I am wrong here.